homeassistant/components/rainmachine/switch.py

```diff
@@ -145,23 +147,59 @@ def unique_id(self) -> str:
             self._rainmachine_entity_id,
         )
 
-    @callback
-    def _program_updated(self):
-        """Update state, trigger updates."""
-        self.async_schedule_update_ha_state(True)
+    async def _async_run_turn_off_coro(self, api_coro) -> None:
+        """Turn the program off."""
+        try:
+            resp = await api_coro
+        except RequestError as err:
+            _LOGGER.error(
+                'Unable to turn off %s "%s": %s',
+                self._switch_type,
+                self.unique_id,
+                err,
+            )
+            return
+
+        if resp["statusCode"] != 0:
+            _LOGGER.error(
+                'Unable to turn off %s "%s": %s',
+                self._switch_type,
+                self.unique_id,
+                resp["message"],
+            )
+            return
+
+        self.hass.async_create_task(self.rainmachine.async_update_programs_and_zones())
+
+    async def _async_run_turn_on_coro(self, api_coro) -> None:
```

Comment:
This method and the one above are very similar. Can we make a helper method that takes the difference as parameters to make it more dry?